Model/Service/Index.php

```diff
@@ -276,6 +304,8 @@ public function handleProductSync(NostoIndexCollection $collection, Store $store
                     $store->getCode()
                 )
             );
+        } catch (MemoryOutOfBoundsException $e) {
```

Comment:
This logic is redundant. If you want this exception to be thrown from this method, simply don't catch it.

Reply:
We need to bubble up the exception in order to stop the indexing process. They must be handled in the `execute` method